sc-memory/tests/sc-memory/common/test_sc_memory_context.cpp

```diff
@@ -293,19 +295,80 @@ void TestActionsUnsuccessfully(std::unique_ptr<ScMemoryContext> const & context,
   TestChangeActionsUnsuccessfully(context, userContext);
 }
 
+TEST_F(ScMemoryTestWithUserMode, GetUserAddrFromContext)
+{
+  ScAddr const & userAddr = m_ctx->CreateNode(ScType::NodeConst);
+
+  TestScMemoryContext userContext{userAddr};
+  EXPECT_EQ(userContext.GetUserAddr(), userAddr);
+}
+
+TEST_F(ScMemoryTestWithUserMode, GetGuestUserAddrFromContext)
+{
+  TestScMemoryContext userContext;
+  EXPECT_TRUE(userContext.GetUserAddr().IsValid());
+}
+
+TEST_F(ScMemoryTestWithUserMode, HandleElementsByGuestUser)
+{
+  TestScMemoryContext userContext;
+  TestActionsUnsuccessfully(m_ctx, userContext);
+}
+
+void TestSetIdentifiedUser(
+    std::unique_ptr<ScMemoryContext> const & context,
+    ScAddr const & guestUserAddr,
+    ScAddr const & userAddr,
+    ScType const & arcType = ScType::EdgeAccessConstPosTemp)
+{
+  ScAddr const & nrelIdentifiedUserAddr{nrel_identified_user_addr};
+  ScAddr const & arcAddr = context->CreateEdge(ScType::EdgeDCommonConst, guestUserAddr, userAddr);
```

Comment:
I think that this arc should be generated by context manager after successful authentication. Otherwise, we will get a situation where an authenticated user generates this arc between another guest and another user and obtains the permissions of the other user